internal/app.go

```diff
@@ -130,6 +214,9 @@ func (a *App) getCleanedCalendar(all []byte) (*ics.Calendar, error) {
 // unfortunate also matches wrong brackets like [MA123) but hey…
 var reTag = regexp.MustCompile(" ?[\\[(](MA|IN|WI|WIB)[0-9]+((_|-|,)[a-zA-Z0-9]+)*[\\])].*")
 
+// matches tags only, and captures the tag in the first group (e.g., "IN0001")
+var reTagOnly = regexp.MustCompile("[[(]((?:MA|IN|WI|WIB|CIT)[0-9]+)(?:(?:_|-|,|\\\\)[ a-zA-Z0-9\\\\]*)*[])]")
```

Comment:
Just noticed: we could also filter event series by `X-CO-RECURRINGID`

Reply:
Good point! Filtering by the whole title would make the URL very long, so I think filtering using the recurring-id would be better. A drawback in this case is that it's not clear for the user what is actually filtered if they look at the URL (as the recurring-id is unknown).
Maybe a good tradeoff would be `<lecture tag>-<recurring id>` (e.g., `IN1234-0123456`)? That way, we don't remove the wrong lectures but the user can still understand the URL

Reply:
I don't think a long url is necessary a bad thing. For instance, this is a valid url used by tum.live: https://edge.live.rbg.tum.de/vod/TensNet_2023_04_24_06_00COMB.mp4/playlist.m3u8?jwt=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2ODI1NzA2NjgsIlVzZXJJRCI6MCwiUGxheWxpc3QiOiJodHRwczovL2VkZ2UubGl2ZS5yYmcudHVtLmRlL3ZvZC9UZW5zTmV0XzIwMjNfMDRfMjRfMDZfMDBDT01CLm1wNC9wbGF5bGlzdC5tM3U4IiwiRG93bmxvYWQiOmZhbHNlLCJTdHJlYW1JRCI6IjI4OTU2IiwiQ291cnNlSUQiOiI1NDYifQ.GvJGAhLMV3OOpp2KzZTCX8aw9jlRNAeH6rC19PsRFhRVvl9rrAX4sWDDyouF7o7Ca5l-QHB2Y5XVWcWnwOiT9QVh33nYGZchJDvJzTxf079V7oVcsac2Q2K_8C5TnNSHNenBe7ISvUXstVZEYWNx4d7S2QYGwFLazSL47qVR9BNkdrBT7F7osw_vpSTTOQwcs65t6vbKVo_CyNzGS5HH0sQjeJGCZVMDqFU_oBxwy4SU9b9bdx8AJJDe8WT3m1UJNk-hcF0YmRqJpcB33XRNZZ_ZP-3hyuuFWvK7BXjFtEQeEVgmyDcLVdFhhVZdf-5VO_RKWnyROGZNLISad55x6w

I think giving the user a list of event names that they can select to be ignored and having them in the url would be a quite intuitive thing to do. Especially because most users won't deselect more than a couple courses.

Reply:
Okay, valid point. Now it's filtering based on the full course name. Please have another look.